### DISCUSSION TITLE
Questions regarding Expr Pro.

### DISCUSSION THREAD

**Discussion by PranavPeshwe** [antonmedv/expr#533] [General] [answered]
Hey @antonmedv ,
Regarding licensing for Expr Pro you mention [here](https://expr-lang.org/expr-pro#license) that:
> The code for Expr Pro is also licensed under the MIT license.

Just wish to make sure I understand this correctly.
- Given that it is MIT licensed, someone could buy it and just put it on GitHub for the public to download (and potentially modify). Is that what you want? While that would be a generous move on your part, it might impact the revenue you would earn from the Pro purchases. I mean, why would people buy something that is available for free 100% legally.
- What about updates and fixes? Do you plan to maintain the Pro code and provide updates? And is that service included in the purchase cost? For how long in the future?

Regards,
Pranav

**Comment by antonmedv** [comment, marked as answer]
Regarding the MIT License for Expr Pro, I find it to be an excellent choice due to its widespread recognition and simplicity, which makes it more accessible to developers. My intention is to make Expr Pro available for use in open-source projects, and opting for the MIT license simplifies things compared to creating a proprietary commercial license. However, I am aware of the limitations, particularly concerning the redistribution of my code on platforms like GitHub. To address this, I am considering the possibility of adapting a commercial license in the future. I acknowledge that those unwilling to pay for software might not do so regardless of the licensing terms. An alternative approach could be to open-source the code under the AGPL license while offering an additional, purchasable license. Yet, I find the idea of selling closed-source software more straightforward. For the time being, I will continue using the MIT license for its ease of use, but I'm open to revisiting this decision later.

As for updates, I am committed to maintaining Expr Pro and providing updates. I believe offering one year of updates post-purchase is fair, but I am open to negotiating different terms if necessary.

**Comment by PranavPeshwe** [comment]
@antonmedv The MIT license is irrevocable. Once something is released with MIT and people download the thing (whether they purchased it or got it for free is immaterial here), you cannot restrict them from distributing it further (thanks to the MIT license). Better to decide on the license thoughtfully before you share the code with anybody, than revisiting it later.
Just my 2 cents.

**Reply by antonmedv** [reply]
This is a good point! I'll think more about this.

**Reply by PranavPeshwe** [reply]
Yup, please do! You've put a considerable amount of effort into developing expr and it is only fair if you select a license for expr Pro that earns you returns. Those who cannot afford the price always have the option to continue to use the non-Pro expr.
I think you should look for how other opensource library owner/maintainers have approached this problem.
